scripts/env-doctor.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+set -e
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
+
+# Function to run a diagnostic and its remedy
+run_diagnostic() {
+  local description="$1"
+  local diagnostic="$2"
+  local remedy_description="$3"
+  local remedy="$4"
+
+  echo -e "${YELLOW}Diagnostic:${NC} $description"
+  if eval "$diagnostic"; then
+    echo -e "${GREEN}✓ Passed${NC}"
+  else
+    echo -e "${RED}✗ Failed${NC}"
+    echo -e "${YELLOW}Remedy:${NC} $remedy_description"
+    if [[ $AUTO_YES != "true" ]]; then
+      read -p "Do you want to apply this remedy? (Yes/No/All) " answer
```

Comment:
Do you think we should explicitly initialize `AUTO_YES` with `false` at the start of the script?